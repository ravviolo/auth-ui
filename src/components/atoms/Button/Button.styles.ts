import styled from 'styled-components';

import { colors } from 'theme';

export const Button = styled.button`
  width: 100%;
  padding: 0.7rem 1rem;
  background-color: ${colors.primary};
  color: ${colors.textWhite};
  letter-spacing: 0.5px;
  text-transform: uppercase;
  border: none;
  border-radius: 8px;
  box-shadow: 2px 2px 8px 1px rgba(0, 0, 0, 0.3);
  cursor: pointer;
  transition: background-color 0.3s ease;

  &:hover {
    background-color: ${colors.primaryDark};
  }
`;
