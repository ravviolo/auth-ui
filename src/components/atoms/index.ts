export { Button } from './Button/Button';
export { Checkbox } from './Checkbox/Checkbox';
export { Divider } from './Divider/Divider';
export { Heading } from './Heading/Heading';
export { IconButton } from './IconButton/IconButton';
export { InputField } from './InputField/InputField';
export { InputLabel } from './InputLabel/InputLabel';
export { TextButton } from './TextButton/TextButton';
