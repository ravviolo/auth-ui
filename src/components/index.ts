export * from './templates/LoginTemplate/LoginTemplate';
export * from './templates/SignUpTemplate/SignUpTemplate';
