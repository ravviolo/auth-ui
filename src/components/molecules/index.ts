export { AuthFooter } from './AuthFooter/AuthFooter';
export { AuthHeader } from './AuthHeader/AuthHeader';
export { AuthSocials } from './AuthSocials/AuthSocials';
export { EmailInput } from './EmailInput/EmailInput';
export { PasswordInput } from './PasswordInput/PasswordInput';
