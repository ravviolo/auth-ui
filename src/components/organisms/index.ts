export * from './LoginForm/LoginForm';
export * from './SignUpForm/SignUpForm';
