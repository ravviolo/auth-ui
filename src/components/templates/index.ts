export * from './LoginTemplate/LoginTemplate';
export * from './SignUpTemplate/SignUpTemplate';
