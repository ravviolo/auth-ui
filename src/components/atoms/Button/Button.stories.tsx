import { ComponentMeta, ComponentStory } from '@storybook/react';

import { Button } from 'components/atoms';

export default {
  title: 'Atoms/Button',
  component: Button,
  args: {
    type: 'button',
    label: 'Button',
  },
} as ComponentMeta<typeof Button>;

export const Default: ComponentStory<typeof Button> = (args) => (
  <Button {...args} />
);
Default.storyName = 'Button';
