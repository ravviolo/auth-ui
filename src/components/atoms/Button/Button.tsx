import * as S from './Button.styles';

interface Props {
  label: string;
  type: 'button' | 'submit' | 'reset';
  onClick?: () => void;
}

export const Button = ({ label, type = 'button', onClick }: Props) => {
  return (
    <S.Button type={type} onClick={onClick}>
      {label}
    </S.Button>
  );
};
