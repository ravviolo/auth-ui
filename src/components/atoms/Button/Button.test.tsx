import { render, screen } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import { create } from 'react-test-renderer';

import { Button } from 'components/atoms';

import { Props as ButtonProps } from './Button';

describe('Button', () => {
  const mockOnClick = jest.fn();

  const Props: ButtonProps = {
    label: 'Test Button',
    testId: 'test-btn',
    type: 'button',
    onClick: mockOnClick,
  };

  it('displays text received via props', () => {
    render(<Button {...Props} />);

    const btn = screen.getByTestId('test-btn');

    expect(btn).toHaveTextContent('Test Button');
  });

  it('handles click events', async () => {
    render(<Button {...Props} />);

    const btn = screen.getByTestId('test-btn');

    await userEvent.click(btn);

    expect(mockOnClick).toHaveBeenCalledTimes(1);
  });

  it('matches snapshot', () => {
    const tree = create(<Button {...Props} />).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
