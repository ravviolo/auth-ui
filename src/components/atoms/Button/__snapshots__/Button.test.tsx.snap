// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button should match snapshot 1`] = `
<button
  className="sc-bczRLJ jXOWFG"
  data-testid="btn-test-id"
  onClick={[MockFunction]}
  type="button"
>
  Test Button
</button>
`;
